### PR TITLE
vscode-utils: expose `toExtensionJson`

### PR DESCRIPTION
**Pull request by ncfavier**
Needed to avoid duplication in https://github.com/nix-community/home-manager/pull/3588#discussion_r1068076130.

### CODE REVIEW

File: pkgs/applications/editors/vscode/extensions/vscode-utils.nix [hunk 1]
```diff
@@ -100,10 +99,42 @@ let
...
+      id = identifier.uuid;
+      publisherId = "";
+      publisherDisplayName = ext.vscodeExtPublisher;
+      targetPlatform = "undefined";
+      isApplicationScoped = false;
+      updated = false;
+      isPreReleaseVersion = false;
+      installedTimestamp = 0;
+      preRelease = false;
+    };
```
Comment (ncfavier) [inline pkgs/applications/editors/vscode/extensions/vscode-utils.nix:129]: Do we actually need all these fields? For reference, this is what a typical entry generated by vscode looks like:

```json
  {
    "identifier": {
      "id": "james-yu.latex-workshop"
    },
    "version": "9.2.1",
    "location": {
      "$mid": 1,
      "fsPath": "/home/n/.vscode-oss/extensions/James-Yu.latex-workshop",
      "path": "/home/n/.vscode-oss/extensions/James-Yu.latex-workshop",
      "scheme": "file"
    }
  }
```

Comment (eadwu) [reply pkgs/applications/editors/vscode/extensions/vscode-utils.nix]: I am on insiders so it may so that it generated a different schema.

Comment (ncfavier) [reply pkgs/applications/editors/vscode/extensions/vscode-utils.nix]: Good point. Could you test this PR on insiders?

Comment (ncfavier) [reply pkgs/applications/editors/vscode/extensions/vscode-utils.nix]: I've dropped that commit for now since it's not needed for this PR

**Pull request merged**